src/doc_scan_service/doc.scan.service.js

```diff
@@ -0,0 +1,204 @@
+const { Payload } = require('../request/payload');
+const Validation = require('../yoti_common/validation');
+const config = require('../../config');
+const Media = require('../data_type/media');
+const DocScanError = require('./doc.scan.error');
+
+/**
+ * @param {string} sessionId
+ *
+ * @returns {string}
+ */
+const sessionPath = sessionId => `/sessions/${sessionId}`;
+
+/**
+ * @param {string} sessionId
+ * @param {string} mediaId
+ *
+ * @returns {string}
+ */
+const mediaContentPath = (sessionId, mediaId) => `${sessionPath(sessionId)}/media/${mediaId}/content`;
+
+/**
+ * Service built to handle the interactions between the client and Doc Scan API's
+ *
+ * @class DocScanService
+ */
+class DocScanService {
+  /**
+   * @param {string} sdkId
+   * @param {string|Buffer} pem
+   */
+  constructor(sdkId, pem) {
+    Validation.isString(sdkId, 'sdkId');
+    Validation.notNullOrEmpty(pem, 'pem');
+
+    this.sdkId = sdkId;
+    this.pem = pem;
+  }
+
+  /**
+   * Uses the supplied session specification to create a session
+   *
+   * @param {SessionSpecification} sessionSpecification
+   *
+   * @returns {Promise} Resolves CreateSessionResult
+   */
+  createSession(sessionSpecification) {
+    Validation.instanceOf(sessionSpecification, SessionSpecification, 'sessionSpecification');
+
+    const request = new RequestBuilder()
+      .withPemString(this.pem)
+      .withBaseUrl(config.yoti.docScanApi)
+      .withEndpoint('/sessions')
+      .withQueryParam('sdkId', this.sdkId)
+      .withPost()
+      .withPayload(new Payload(sessionSpecification))
+      .withHeader('Content-Type', 'application/json')
+      .build();
+
+    return new Promise((resolve, reject) => {
+      request.execute()
+        .then((response) => {
+          try {
+            return resolve(new CreateSessionResult(response.getParsedResponse()));
+          } catch (err) {
+            return reject(new DocScanError(err));
+          }
+        })
+        .catch(err => reject(new DocScanError(err)));
+    });
+  }
+
+  /**
+   * Retrieves the current state of a given session
+   *
+   * @param {string} sessionId
+   *
+   * @returns {Promise} Resolves GetSessionResult
+   */
+  getSession(sessionId) {
+    Validation.isString(sessionId, 'sessionId');
+
+    const request = new RequestBuilder()
+      .withPemString(this.pem)
+      .withBaseUrl(config.yoti.docScanApi)
+      .withEndpoint(sessionPath(sessionId))
+      .withQueryParam('sdkId', this.sdkId)
+      .withGet()
+      .build();
+
+    return new Promise((resolve, reject) => {
+      request.execute()
+        .then((response) => {
+          try {
+            return resolve(new GetSessionResult(response.getParsedResponse()));
+          } catch (err) {
+            return reject(new DocScanError(err));
+          }
+        })
+        .catch(err => reject(new DocScanError(err)));
+    });
+  }
+
+  /**
+   * Deletes a session and all of it's associated content
```

Comment:
```suggestion
   * Deletes a session and all of its associated content
```
I realised these two apostrophes are present in Java too. 
@MrBurtyyy no rush, but please could you update them in Java too?

Reply:
Corrected in 0b91cfd